pandas/core/internals/blocks.py

```diff
@@ -1297,24 +1296,20 @@ def take_nd(self, indexer, axis, new_mgr_locs=None, fill_tuple=None):
 
         if fill_tuple is None:
             fill_value = self.fill_value
-            new_values = algos.take_nd(
-                values, indexer, axis=axis, allow_fill=False, fill_value=fill_value
-            )
+            allow_fill = False
         else:
             fill_value = fill_tuple[0]
-            new_values = algos.take_nd(
-                values, indexer, axis=axis, allow_fill=True, fill_value=fill_value
-            )
+            allow_fill = True
 
+        new_values = algos.take_nd(
+            values, indexer, axis=axis, allow_fill=allow_fill, fill_value=fill_value
+        )
+
+        # Called from three places in managers, all of which satisfy
+        #  this assertion
+        assert not (axis == 0 and new_mgr_locs is None)
```

Comment:
Can you raise a ValueError here instead of using assert? In the off chance someone uses -O flag assert won't help

Reply:
asserts are ok in the internals